libraries/search/lib/objective/ExceptionObjectiveFunction.ts

```diff
@@ -32,22 +31,25 @@ export class ExceptionObjectiveFunction<
 > extends ObjectiveFunction<T> {
   protected _error: Error;
 
-  constructor(subject: SearchSubject<T>, id: string, error: Error) {
-    super(id, subject);
+  constructor(id: string, error: Error) {
+    super(id);
     this._error = error;
   }
 
   get error() {
     return this._error;
   }
 
-  /**
-   * @inheritDoc
-   */
-  calculateDistance(): number {
-    // This method should never be called.
-    // The exception objective function is only created when an exception is already covered.
-    // So the distance is always zero.
-    throw new Error(shouldNeverHappen("method not implemented."));
+  override calculateDistance(encoding: T): number {
```

Comment:
How would this calculation work for an exception?

Reply:
We only check whether the ID is covered. so we can still give a value back. Point is that i need this function to always provide a value instead of throwing an error.